### PR TITLE
feat: lock sBTC when initializing withdrawal

### PR DESCRIPTION
**Pull request by hstove**
This uses the new sBTC token contract to properly lock the user's sBTC when they initialize a withdrawal request.

### CODE REVIEW

**Review by setzeus — CHANGES_REQUESTED** [review]
Overall looks great, maybe just add a test to cover withdrawing less than the dust amount? Once that's included will be good to merge.

**Comment by hstove** [comment]
Good call @setzeus , ready for re-review!

**Review by setzeus — APPROVED** [review]

File: contracts/tests/sbtc-withdrawal.test.ts [hunk 1]
```diff
@@ -150,12 +200,33 @@ describe("initiating a withdrawal request", () => {
...
+          amount: 1000n,
           recipient: newPoxAddr(6, new Uint8Array(20)),
           maxFee: 10n,
         }),
         alice
       ).value
     ).toEqual(errors.withdrawal.ERR_INVALID_ADDR_HASHBYTES);
   });
+
+  test("Cannot try and withdrawal less than or equal to the dust limit", () => {
```
Comment (friedger) [inline contracts/tests/sbtc-withdrawal.test.ts:212]: nit: To match the pattern above "test withdrawal amount of less than or equal to the dust limit is rejected"

File: contracts/contracts/sbtc-withdrawal.clar [hunk 2]
```diff
@@ -1,9 +1,13 @@
 ;; Error codes
 
+(define-constant dust-limit u546)
```
[the remark this answers is not in the document]
Comment (hstove) [reply contracts/contracts/sbtc-withdrawal.clar:3]: Yeah, the dust limit is a constant (it matches the BTC dust limit). In this case, it's the minimum amount you can withdrawal. Fees are dynamic, but they're on top of your "withdrawal amount", so this dust limit is still applicable

**Comment by setzeus** [comment]
@friedger I fixed the broken test, can you re-review when you have a few mins here? Hank is OOO for the next few weeks & I'd like to get this merged in asap.

File: contracts/contracts/sbtc-withdrawal.clar (hunk 2 above)
[the remark this answers is not in the document]
Comment (friedger) [reply contracts/contracts/sbtc-withdrawal.clar]: Who pays for the fees of withdrawal?
What happens when I want to withdraw dust + 1 sat?

**Comment by radicleart** [comment]
Any links to the design docs to get a feel for the overall flow ?

**Comment by radicleart** [comment]
The token contract could explicitly declare impl-trait for sip 10 ? What are the guidelines on this? 

Also the token name is "sBTC Mini" is this right ?

**Review by radicleart — COMMENTED** [review]
Looks good !

**Comment by setzeus** [comment]
> Any links to the design docs to get a feel for the overall flow ?

Yeap! https://docs.google.com/document/d/1Bwi1jM_7p9OqNGT41eWCuoKnfoOrNHycqxby3XSonCA/edit?usp=sharing

**Comment by setzeus** [comment]
> The token contract could explicitly declare impl-trait for sip 10 ? What are the guidelines on this?
> 
> Also the token name is "sBTC Mini" is this right ?

Ah, thank you, caught & updated.

**Review by radicleart — APPROVED** [review]
Looks good .

**Pull request merged**